core/src/main/java/io/kestra/core/plugins/serdes/PluginDeserializer.java

```diff
@@ -71,38 +78,60 @@ private void checkState() {
             }
         }
     }
-    
+
     @SuppressWarnings("unchecked")
     private T fromObjectNode(JsonParser jp,
                              JsonNode node,
                              DeserializationContext context) throws IOException {
         Class<? extends Plugin> pluginType = null;
-        
+
         final String identifier = extractPluginRawIdentifier(node);
         if (identifier != null) {
             log.trace("Looking for Plugin for: {}",
                 identifier
             );
             pluginType = pluginRegistry.findClassByIdentifier(identifier);
         }
-        
+
         if (pluginType == null) {
             String type = Optional.ofNullable(identifier).orElse("<null>");
             throwInvalidTypeException(context, type);
         } else if (Plugin.class.isAssignableFrom(pluginType)) {
             log.trace("Read plugin for: {}",
                 pluginType.getName()
             );
+
+            if (DataChart.class.isAssignableFrom(pluginType)) {
```

Comment:
Because there is too much genericity, basically each data type has a base column descriptor definition, their own one + one that is dependant on the graph type and our schema generator struggles with that much genericity since JSON Schemas need predetermined paths for its resolution. It means that it should start off generic with multiple possible schemas and restrict them as we move, but suddenly when typing `type: ...Executions` for the datafilter type it has to swap to a schema based on both the chart type and datafilter type.
If you have a look at the generated schema, definitions are each composed of all the types to have the most specific one each time 
![image](https://github.com/user-attachments/assets/1b1e423f-43ad-4a0c-8c92-afdec284e66f)